python/python/lance/indices.py

```diff
@@ -0,0 +1,253 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright The Lance Authors
+import math
+import warnings
+from typing import TYPE_CHECKING, Optional, Union
+
+import pyarrow as pa
+
+from lance.file import LanceFileReader, LanceFileWriter
+
+from .lance import indices
+
+if TYPE_CHECKING:
+    from .dependencies import torch
+
+
+class IvfModel:
+    """
+    A class that represents a trained IVF model.
+    """
+
+    def __init__(self, centroids: pa.Array, distance_type: str):
+        self.centroids = centroids
+        """The centroids of the IVF clusters"""
+        self.distance_type = distance_type
+        """The distance type used to train the IVF model"""
+
+    def num_partitions(self) -> int:
```

Comment:
Seems like this should be a property:
```suggestion
    @property
    def num_partitions(self) -> int:
```

Reply:
Changed to a property